File: src/app/core/types/catalogue.ts
export class Catalogue {
  id: number;
  title: string;
  desc: string;
  version: string;
  file: string|any;
  fileUrl: string;
  fileExtension: string;

  constructor(id: number, title: string, desc: string, version: string, file: string|any, fileUrl: string, fileExtension: string) {
    this.id = id;
    this.title = title;
    this.desc = desc;
    this.version = version;
    this.file = file;
    this.fileUrl = fileUrl;
    this.fileExtension = fileExtension;
  }
}

File: backend/Marbomil.Infrastructure/Entities/Database/Catalogue.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Marbomil.Infrastructure.Entities.Database
{
    public class Catalogue : IDbEntity
    {
        public int Id { get; set; }

        [MaxLength(100)]
        public string Title { get; set; }

        public string Description { get; set; }

        [MaxLength(100)]
        public string Version { get; set; }

        [MaxLength(1000)]
        public string FileUrl { get; set; }
    }
}
